doc/book/zend.authentication.intro.md

```diff
@@ -0,0 +1,408 @@
+
+## Adapters
+
+`Zend\Authentication` adapters are used to authenticate against a particular type of authentication
+service, such as *LDAP*, *RDBMS*, or file-based storage. Different adapters are likely to have
+vastly different options and behaviors, but some basic things are common among authentication
+adapters. For example, accepting authentication credentials (including a purported identity),
+performing queries against the authentication service, and returning results are common to
+`Zend\Authentication` adapters.
+
+Each `Zend\Authentication` adapter class implements `Zend\Authentication\Adapter\AdapterInterface`.
+This interface defines one method, `authenticate()`, that an adapter class must implement for
+performing an authentication query. Each adapter class must be prepared prior to calling
+`authenticate()`. Such adapter preparation includes setting up credentials (e.g., username and
+password) and defining values for adapter-specific configuration options, such as database
+connection settings for a database table adapter.
+
+The following is an example authentication adapter that requires a username and password to be set
+for authentication. Other details, such as how the authentication service is queried, have been
+omitted for brevity:
+
+```php
+use Zend\Authentication\Adapter\AdapterInterface;
+
+class My\Auth\Adapter implements AdapterInterface
+{
+    /**
+     * Sets username and password for authentication
+     *
+     * @return void
+     */
+    public function __construct($username, $password)
+    {
+        // ...
+    }
+
+    /**
+     * Performs an authentication attempt
+     *
+     * @return \Zend\Authentication\Result
+     * @throws \Zend\Authentication\Adapter\Exception\ExceptionInterface
+     *               If authentication cannot be performed
+     */
+    public function authenticate()
+    {
+        // ...
+    }
+}
+```
+
+As indicated in its docblock, `authenticate()` must return an instance of
+`Zend\Authentication\Result` (or of a class derived from `Zend\Authentication\Result`). If for some
+reason performing an authentication query is impossible, `authenticate()` should throw an exception
+that derives from `Zend\Authentication\Adapter\Exception\ExceptionInterface`.
+
+## Results
+
+`Zend\Authentication` adapters return an instance of `Zend\Authentication\Result` with
+`authenticate()` in order to represent the results of an authentication attempt. Adapters populate
+the `Zend\Authentication\Result` object upon construction, so that the following four methods
+provide a basic set of user-facing operations that are common to the results of
+`Zend\Authentication` adapters:
+
+-   `isValid()`- returns `TRUE` if and only if the result represents a successful authentication
+attempt
+-   `getCode()`- returns a `Zend\Authentication\Result` constant identifier for determining the type
+of authentication failure or whether success has occurred. This may be used in situations where the
```

Comment:
Lines that belong to a bullet point need to be indented two spaces. (Again, may need to do a post-processor.)